### ISSUE TITLE
Recuperação de Arquivos

### ISSUE THREAD

**Issue by euviniciusmendonca** [supergovbr/mod-sei-pen#205]
**Sua solicitação de melhoria está relacionada a um problema? Por favor descreva.**
Uma gestora reportou que processos recebidos por meio do Tramita.GOV.BR estavam com problema no carregamento de documentos:

![WhatsApp Image 2023-01-10 at 14 57 09](https://user-images.githubusercontent.com/60577185/211648219-7e048677-81da-42a9-b124-c0e17b3f135a.jpeg)

O problema supracitado não tem relação com o trâmite e muito menos com o Tramita.GOV.BR. 

Identificou-se uma inconsistência interna na própria infraestrutura do órgão, que salvou os arquivos dos processos recebidos em outra localidade fora do padrão estabelecido, no docker, e posteriormente sendo excluídos quando o container foi apagado.

----

**Descreva a solução que você gostaria**

A recuperação desses documentos por meio do módulo (Consultando o banco de dados do Tramita.GOV.BR ???)

**Comment by guilhermeadc** [comment]
Será necessário a imp

**Comment by MatheusMarques80** [comment]
PRs da demanda: https://github.com/supergovbr/super/pull/225 e https://github.com/supergovbr/mod-sei-pen/pull/241

**Comment by MatheusMarques80** [comment]
Resolvido os conflitos que estavam dando com a branch master

**Comment by euviniciusmendonca** [comment]
@higodf  @caduvieira @guilhermeadc Ao tentar o envio externo de processo é gerado o seguinte erro:

![image](https://github.com/supergovbr/mod-sei-pen/assets/60577185/0e192bd9-0815-4b26-a0b7-2f8a50ced0ff)

**Comment by davidnascimento1618** [comment]
@euviniciusmendonca, testando localmente, notou-se que o comportamento parece estar condizente com o que foi solicitado, como mostra no _.gif_  abaixo:

![Exibição do botão de 'Reenvio Externo de Processo' quando o processo está bloqueado](https://github.com/supergovbr/mod-sei-pen/assets/87980947/8ac8d04f-ce5b-4ade-b08b-6acb8b86618a)

Confirma o comportamento normal ou este botão só deveria ser exibido em algum _status_ específico? Pergunto porque o processo pode só estar bloqueado porque ainda não chegou no Tramita e não porque houve alguma falha no caminho, nesse comportamento atual, no qual foi implementado, o usuário pode (nesse meio tempo) ficar reenviando a todo instante, já que o botão ficará sempre disponível enquanto ele estiver bloqueado.

@guilhermeadc, possa ser que a _release_, a qual foi disponibilizada essa melhoria, veio com conflitos ou mesmo abarcou apenas um dos dois _pull requests_: 225 (Super) e 241 (mod-sei-pen), por tal motivo, está ocasionando este erro relatado pelo Vinícius, no ambiente disponibilizado, ou seja, no [órgão 6](http://super.orgao6.tramita.processoeletronico.gov.br).

**Comment by davidnascimento1618** [comment]
@MatheusMarques80,

Algumas regras a serem consideradas e reforçadas:

**Regra 1:** para o botão 'Reenvio Externo de Processo' aparecer, o processo deverá estar bloqueado e o _status_ da tramitação externa como: 'Tramitação externa do processo XXXXXX.XXXXXX/XXXX-XX concluída com sucesso' (id_tarefa_modulo='PEN_OPERACAO_EXTERNA');

**Regra 2:** só pode ser exibido o recurso 'pen_reenviar_processo' exclusivamente para o perfil 'Administrador';

**Regra 3:** o ícone do botão de 'Reenvio Externo de Processo' deverá ser alterado, o novo ícone será repassado pela CGSIS;

**Regra 4:** ao realizar o reenvio para o remetente do processo, o módulo deverá alimentar o registro de documentos imutáveis deste processo, caso ainda não exista um registro para estes documentos. 
**OBS.:** reunir com a equipe técnica da CGSIS para maiores orientações.

**Comment by davidnascimento1618** [comment]
@euviniciusmendonca, conforme supracitado na ['Regra 3'](https://github.com/supergovbr/mod-sei-pen/issues/205#issuecomment-1574292541-id3), o ícone não está tão intuitivo para o usuário, sugiro a solicitação de criação (para a Mônica) de um novo ícone desta funcionalidade em questão, para que seja repassado para equipe da _First_.

Atualmente o ícone de 'Reeenvio Externo de Processo' é este, conforme a ilustração abaixo:

![image](https://github.com/supergovbr/mod-sei-pen/assets/87980947/6739de61-957b-4c0b-b221-93219d260ce9)

Ou seja, encontra-se em duplicidade com o ícone de 'Envio Externo de Processo', como você pode observar abaixo:

![image](https://github.com/supergovbr/mod-sei-pen/assets/87980947/6ac7a099-8737-4e56-8ef6-d34523c5e01e)

**Comment by davidnascimento1618** [comment]
Segue em anexo (em arquivo _.svg_) o ícone referente tanto ao 'Envio Externo de Processo', quanto para o 'Reenvio Externo de Processo', ficou definido que ambos servirão para as duas funcionalidades.

![envio_reenvio_externo_de_processo](https://github.com/supergovbr/mod-sei-pen/assets/87980947/0834c1dc-a41c-4f34-bfaa-93897c779e71)

**Comment by davidnascimento1618** [comment]
Em reunião com a equipe negocial, foi acertado que o botão de 'Reenvio Externo de Processo', só será exibido no _status_ **6** no Tramita ("6 - Recibo de conclusão do trâmite recebido pelo remetente".) e não mais apenas se o processo estiver bloqueado.

**Comment by MatheusMarques80** [comment]
Ao realizar o reenvio externo, o processo sempre volta como recusado, segue abaixo os passos realizados assim como uma imagem do resultado.

1° Realizar um envio externo do processo e aguardar a finalização do trâmite (Situação: Trâmite do processo XXXXXXXXXXX foi concluído)

2° Para exibição do botão de reenvio externo o processo deve está como bloqueado e com a situação do trâmite mencionada no primeiro passo.

3° Realizar o Reenvio externo do processo.

Imagem com o resultado final da realização dos passos acima

![Image](https://github.com/supergovbr/mod-sei-pen/assets/13146046/523f34d2-478f-407d-aef3-b001efb07398)

**Comment by MatheusMarques80** [comment]
@davidnascimento1618 Sim, foi aplicada a regra de número 4 e foi realizado o teste com o script mencionado.
